src/modules/interactive.js

```diff
@@ -0,0 +1,23 @@
+export const clearCompleted = (list, updateIndexFunc) => {
+  const completedTasks = document.querySelectorAll('.completed');
+  completedTasks.forEach((task) => {
+    list.tasks = list.tasks.filter((item) => (item.index !== Number(task.getAttribute('key'))));
+    task.remove();
+  });
+
+  updateIndexFunc();
+};
+
+export const changeCheckbox = (checkbox, task, updateStorageFunc, list) => {
+  checkbox.addEventListener('click', () => {
+    if (checkbox.checked === true) {
+      task.classList.add('completed');
+      list.tasks[Number(task.getAttribute('key'))].completed = true;
+      updateStorageFunc();
+    } else {
+      task.classList.remove('completed');
+      list.tasks[Number(task.getAttribute('key'))].completed = false;
+      updateStorageFunc();
+    }
+  });
+};
```

Comment:
- [ ] 👍 You've done a great job implementing the required functions for updating task object values and handling the "Clear all completed" button using the Array.filter() method. However, there is an issue that needs to be addressed:

  - When the page loads and there are no todos stored in the local storage, there is an error that prevents users from adding new todos to your webpage. This error occurs because you are using a forEach loop on the null property. To resolve this issue, ensure that your functionality works perfectly even if the local storage is empty. Currently, you have two functions that meet the project requirement but I am unable to check the functionality in the webpage. 

[To-Do-List.webm](https://github.com/Movgto/ToDoList/assets/55840999/675eb1bf-cc1d-4bf3-8eb8-e5239add1887)

By handling this error, your project will function smoothly and allow users to add new todos without any hindrance and will be able to review your code for this milestone. Keep up the good work, and I'm excited to see your project working flawlessly once this issue is resolved! 🌟📝🚀

Reply:
@shahadat3669 Thank you for your feedback, but the problem is that I'm not sure how to approach this issue because I've tested the application from my side and it's working correctly without any errors as you can see in the following video:
[Todolist app testing](https://www.loom.com/share/c710cc342ece468d894f11d5189e1172?sid=a40854f3-5863-4e8a-b346-3efc077304d7)
Please can you help with this?

Reply:
@Movgto Greate job! 👍